VMEncryption/main/ResourceDiskUtil.py

```diff
@@ -0,0 +1,274 @@
+        return (int)(self.executor.ExecuteInBash(cmd, suppress_logging=True)) == 0
+
+    def is_valid_key(self):
+        """ test if current key can be used to open current partition """
+        # suppress logging to avoid log clutter if the key doesn't match
+        if not self.resource_disk_partition_exists():
+            return False
+        cmd = 'cryptsetup luksOpen ' + self.RD_DEV_PATH + ' --test-passphrase --key-file ' + self.RD_KEY_FILE
+        return (int)(self.executor.Execute(cmd, suppress_logging=True)) == 0
+
+    def resource_disk_exists(self):
+        """ true if the udev name for resourced disk exists """
+        cmd = 'test -b ' + self.RD_BASE_DEV_PATH
+        return (int)(self.executor.Execute(cmd, suppress_logging=True)) == 0
+
+    def resource_disk_partition_exists(self):
+        """ true if udev name for resource disk partition exists """
+        cmd = 'test -b ' + self.RD_DEV_PATH
+        return (int)(self.executor.Execute(cmd, suppress_logging=True)) == 0
+
+    def format_luks(self):
+        """ set up resource disk crypt device layer using disk util """
+        if not self.resource_disk_partition_exists():
+            self.logger.log('LUKS format operation requested, but resource disk partition does not exist')
+            return False
+        return (int)(self.disk_util.luks_format(passphrase_file=self.RD_KEY_FILE, dev_path=self.RD_DEV_PATH, header_file=None)) == 0
+
+    def encrypt(self):
+        """ use disk util with the appropriate device mapper """
+        self.mount_key_volume()
+        return (int)(self.disk_util.encrypt_disk(dev_path=self.RD_DEV_PATH, passphrase_file=self.RD_KEY_FILE, mapper_name=self.mapper_name, header_file=None)) == 0
+
+    def make(self):
+        """ make a default file system on top of the crypt layer """
+        make_result = self.disk_util.format_disk(dev_path=self.mapper_path, file_system=CommonVariables.default_file_system)
+        if make_result != CommonVariables.process_success:
+            self.logger.log(msg="Failed to make file system on ephemeral disk", level=CommonVariables.ErrorLevel)
+            return False
+        # todo - drop DATALOSS_WARNING_README.txt file to disk
+        return True
+
+    def mount_key_volume(self):
+        """ attempt to mount the key volume and verify existence of key file"""
+        if not os.path.exists(self.RD_KEY_FILE):
+            self.disk_util.make_sure_path_exists(self.RD_KEY_FILE_MOUNT_POINT)
+            key_volume_device_name = os.popen('blkid -L "' + self.RD_KEY_VOLUME_LABEL + '"').read().strip()
+            self.disk_util.mount_filesystem(key_volume_device_name, self.RD_KEY_FILE_MOUNT_POINT)
+        return os.path.exists(self.RD_KEY_FILE)
+        
+    def mount(self):
+        """ mount the file system previously made on top of the crypt layer """
+        #ensure that resource disk mount point directory has been created
+        cmd = 'mkdir -p ' + self.RD_MOUNT_POINT
```

Comment:
the disk_util.mount_filesystem() call below already takes care of this.

Reply:
Unfortunately, disk_util.mount_filesystem() ignores the return code of make_sure_path_exists.  I want this function to fail early and log the error if it is unable to create the directory.